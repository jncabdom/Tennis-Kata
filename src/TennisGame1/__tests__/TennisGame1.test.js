const TennisGame1 = require('./TennisGame1');

function addPointPlayer(number, name, game) {
  while (number >= 1) {
    game.wonPoint(name);
    number--;
  }
}

describe('TennisGame1', () => {
  test('should execute the game score OK', () => {
    const game = new TennisGame1('player1', 'player2');
    expect(game.getScore()).toBe('Love-All');
  });

  test('should after 3 point of player1 have a result of Forty-Love', () => {
    const game = new TennisGame1('player1', 'player2');
    game.wonPoint('player1');
    game.wonPoint('player1');
    game.wonPoint('player1');
    expect(game.getScore()).toBe('Forty-Love');
  });

});

describe('All cases', () => {
  test('ANDAMIO', () => {

    const allGameCasesINPUT =
      [
        [0, 0],
        [1, 0],
        [2, 0],
        [3, 0],
        [4, 0],

        [0, 1],
        [0, 2],
        [0, 3],
        [0, 4],

        [1, 1],
        [2, 2],
        [3, 3],
        [4, 4],

        [5, 4],
        [4, 5],
        [6, 4],
        [4, 6]
      ];

    const allGameCasesRESULT = [];

    allGameCasesINPUT.forEach(input => {
      const game = new TennisGame1('player1', 'player2');

      // Player 1
      addPointPlayer(input[0], game.player1Name, game);
      // Player 2
      addPointPlayer(input[1], game.player2Name, game);

      allGameCasesRESULT.push(game.getScore());

    });
    expect(allGameCasesRESULT).toMatchSnapshot();
  });
});
