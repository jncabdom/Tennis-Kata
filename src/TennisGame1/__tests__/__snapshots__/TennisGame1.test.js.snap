// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All cases ANDAMIO 1`] = `
Array [
  "Love-All",
  "Fifteen-Love",
  "Thirty-Love",
  "Forty-Love",
  "Win for player1",
  "Love-Fifteen",
  "Love-Thirty",
  "Love-Forty",
  "Win for player2",
  "Fifteen-All",
  "Thirty-All",
  "Deuce",
  "Deuce",
  "Advantage player1",
  "Advantage player2",
  "Win for player1",
  "Win for player2",
]
`;
